### PR TITLE
add makefile

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
# Blog App Backend:

Description: Added a makefile. 

Installation Instructions:
1. make a virtual environment by running `python3 -m venv env`
2. Open terminal and start virtual environment by running `source env/bin/activate`
3. run `make requirements` to make requirements
4. run `make serve` to start the django server

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -1,2 +1,32 @@
+help:
+	@echo "Please use \`make <target>\` where <target> is one of"
+	@echo "  help                       display this help message"
+	@echo "  migrate                    apply database migrations"
+	@echo "  prod_requirements          install production requirements for edly_panel python3 environment"
+	@echo "  quality                    check code quality for edly_panel"
+	@echo "  requirements               install local requirements for edly_panel python3 environment"
+	@echo "  serve                      serve edly_panel App at 0.0.0.0:9999"
```
Comment (zubair-arbi) [inline makefile]: nit: `serve edly_panel App at 0.0.0.0:9999` -> `serve django-rest-blog-app-backend at 0.0.0.0:8000`

File: makefile [hunk 2]
```diff
@@ -1,2 +1,32 @@
+help:
+	@echo "Please use \`make <target>\` where <target> is one of"
+	@echo "  help                       display this help message"
+	@echo "  migrate                    apply database migrations"
+	@echo "  prod_requirements          install production requirements for edly_panel python3 environment"
+	@echo "  quality                    check code quality for edly_panel"
```
Comment (zubair-arbi) [inline makefile]: nit: `edly_panel` -> `django-rest-blog-app-backend`

File: makefile [hunk 3]
```diff
@@ -1,2 +1,32 @@
...
+	python manage.py collectstatic --noinput
+
+serve:
+	python manage.py runserver 127.0.0.1:8000
+
+migrate:
+	python manage.py migrate --noinput
+
+test:
+	python manage.py test
```
Comment (zubair-arbi) [inline makefile]: We will use `pytest` for writing/running tests

File: makefile [hunk 4]
```diff
@@ -1,2 +1,32 @@
...
+	pip install -qr requirements/local.txt --exists-action w
+
+prod_requirements:
+	pip install -qr requirements/production.txt --exists-action w
+
+static:
+	python manage.py collectstatic --noinput
+
+serve:
+	python manage.py runserver 127.0.0.1:8000
```
Comment (zubair-arbi) [inline makefile]: `127.0.0.1:8000` -> `0.0.0.0:8000` (this way you can access you local site from any computer from same network.)

**Comment by zubair-arbi** [comment]
Update the `README` file as well.

File: makefile (hunk 1 above)
Comment (Ali-D-Akbar) [reply makefile]: done

File: makefile (hunk 2 above)
Comment (Ali-D-Akbar) [reply makefile]: Done.

File: makefile (hunk 3 above)
Comment (Ali-D-Akbar) [reply makefile]: Done. I had to add one line of code to tests.py too in order to use pytest.

File: makefile (hunk 4 above)
Comment (Ali-D-Akbar) [reply makefile]: Done.

**Pull request merged**